### ISSUE TITLE
`poly test` fails when a clj file does not have a ns declaration

### ISSUE THREAD

**Issue by ieugen** [polyfy/polylith#279]
**Describe the bug**

We are using a `my/config.clj` file as a configuration. 
The config.clj does not have a ns declaration. 

The config file is loaded with a modified version of crispin:  https://github.com/dunaj-project/crispin/blob/eff885fc0bf1d850a0daaf08efd4e9a9b061f24e/src/crispin/core.cljc#L105 

The code eventually loads it via `clojure.core/load-string` fn .
But I think the issue might be with Poly test tool trying to process the source files !? 
```
       (let [^java.net.URI puri (if (instance? java.net.URI uri)
                                  uri
                                  (java.net.URI/create uri))
             r (condp = (.getScheme puri)
                 "cp" (jio/resource (.getSchemeSpecificPart puri))
                 uri)
             s (when r (slurp r))]
         (when s
           (log/debug "Load configuration resource" uri))
         (condp = parser
           :json (cc/parse-string s)
           :edn (ce/read-string s)
           :clj (load-string s)))
```

my/config.clj
```
;; (ns my.config)
;; without the ns declaration above, polylith fails with a strange exception 
{:sentry {:dsn ""
          :environment ""}}
```

**To Reproduce**
1. Navigate to workspace directory
2. Run `clojure -M:poly test`
3. See error

Relates to https://github.com/polyfy/polylith/issues/208 ?! 

**Operating System (please complete the following information):**
 - OS:Debian
 - Version: Debian Testing / Bookworm -(12) 

**Versions (please complete the following information):**
 - Java: 17 eclipse temurin
 - Poly: 
```
             :poly             {:extra-paths ["development/test-resoueces"]
                                :jvm-opts    ["-XX:-OmitStackTraceInFastThrow"
                                              ;; Run RCF tests when loading files in REPL
                                              ;; https://github.com/hyperfiddle/rcf#ci
                                              "-Dhyperfiddle.rcf.generate-tests=true"]
                                :main-opts   ["-m" "polylith.clj.core.poly-cli.core"]
                                :extra-deps  {polyfy/polylith {:git/url   "https://github.com/polyfy/polylith"
                                                               ;;     :tag       "v0.2.17-alpha"
                                                               :sha       "a1581ccc87009127f2151579b826258dd57e3077"
                                                               :deps/root "projects/poly"}}}}
```

**Additional context**
```
clojure.lang.ExceptionInfo: Error while evaluating form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) in class-loader. Cause: java.lang.reflect.InvocationTargetException; Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote; clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)} {:form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote )))}
        at polylith.clj.core.test_runner_orchestrator.core$__GT_eval_in_project$fn__9252.invoke(core.clj:110)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements$fn__31730.invoke(core.clj:59)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements.invokeStatic(core.clj:58)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements.invoke(core.clj:52)
        at polylith.clj.core.clojure_test_test_runner.core$create$reify__31757.run_tests(core.clj:98)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project_with_test_runner.invokeStatic(core.clj:88)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project_with_test_runner.invoke(core.clj:77)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project.invokeStatic(core.clj:159)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project.invoke(core.clj:129)
        at polylith.clj.core.test_runner_orchestrator.core$run.invokeStatic(core.clj:214)
        at polylith.clj.core.test_runner_orchestrator.core$run.invoke(core.clj:197)
        at polylith.clj.core.test_runner_orchestrator.interface$run.invokeStatic(interface.clj:5)
        at polylith.clj.core.test_runner_orchestrator.interface$run.invoke(interface.clj:4)
        at polylith.clj.core.command.test$run.invokeStatic(test.clj:11)
        at polylith.clj.core.command.test$run.invoke(test.clj:5)
        at polylith.clj.core.command.core$execute.invokeStatic(core.clj:88)
        at polylith.clj.core.command.core$execute.invoke(core.clj:63)
        at polylith.clj.core.command.interface$execute_command.invokeStatic(interface.clj:5)
        at polylith.clj.core.command.interface$execute_command.invoke(interface.clj:4)
        at polylith.clj.core.poly_cli.core$_main.invokeStatic(core.clj:33)
        at polylith.clj.core.poly_cli.core$_main.doInvoke(core.clj:7)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.main$main_opt.invokeStatic(main.clj:514)
        at clojure.main$main_opt.invoke(main.clj:510)
        at clojure.main$main.invokeStatic(main.clj:664)
        at clojure.main$main.doInvoke(main.clj:616)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.main.main(main.java:40)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at polylith.clj.core.common.class_loader$invoke_in_STAR_.invokeStatic(class_loader.clj:31)
        at polylith.clj.core.common.class_loader$invoke_in_STAR_.doInvoke(class_loader.clj:27)
        at clojure.lang.RestFn.invoke(RestFn.java:494)
        at polylith.clj.core.common.class_loader$eval_in_STAR_$print_read_eval__287.invoke(class_loader.clj:49)
        at polylith.clj.core.common.class_loader$eval_in_STAR_.invokeStatic(class_loader.clj:51)
        at polylith.clj.core.common.class_loader$eval_in_STAR_.invoke(class_loader.clj:45)
        at polylith.clj.core.common.class_loader$eval_in.invokeStatic(class_loader.clj:61)
        at polylith.clj.core.common.class_loader$eval_in.invoke(class_loader.clj:60)
        at polylith.clj.core.common.interface$eval_in.invokeStatic(interface.clj:23)
        at polylith.clj.core.common.interface$eval_in.invoke(interface.clj:22)
        at polylith.clj.core.test_runner_orchestrator.core$__GT_eval_in_project$fn__9252.invoke(core.clj:108)
        ... 30 more
Caused by: Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7132)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.analyze(Compiler.java:6762)
        at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3900)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7126)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.analyze(Compiler.java:6762)
        at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:6137)
        at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5479)
        at clojure.lang.Compiler$FnExpr.parse(Compiler.java:4041)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7122)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.eval(Compiler.java:7191)
        at clojure.lang.Compiler.eval(Compiler.java:7183)
        at clojure.lang.Compiler.eval(Compiler.java:7149)
        at clojure.core$eval.invokeStatic(core.clj:3215)
        at clojure.core$eval.invoke(core.clj:3211)
        at clojure.core$eval3647.invokeStatic(NO_SOURCE_FILE:0)
        at clojure.core$eval3647.invoke(NO_SOURCE_FILE)
        at clojure.lang.Compiler.eval(Compiler.java:7194)
        at clojure.lang.Compiler.eval(Compiler.java:7149)
        ... 45 more
Caused by: clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)}
        at clojure.lang.Compiler$ConstantExpr$Parser.parse(Compiler.java:2016)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7124)
        ... 65 more
Couldn't run test statement for the ontolog project: (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) clojure.lang.ExceptionInfo: Error while evaluating form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) in class-loader. Cause: java.lang.reflect.InvocationTargetException; Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote; clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)} {:form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote )))}

Test results:  passes,  failures,  errors.
```

**Comment by tengstrand** [comment]
Hi @ieugen!

I think this has been solved by issue #259 which is merged to master.
Please try it out with the latest sha from master: e4d3f0e67bf1082d686b3049682b3f811cb1925a

**Comment by tengstrand** [comment]
Is this still a problem @ieugen ?

**Comment by ieugen** [comment]
Hi, I just tested this again and it's still failing. 

What I did:
- ran `clojure -M:poly test` - all tests pass.
- edited file `dre/app/config.clj` and commented out `;; (ns dre.app.config)` 
- ran `clojure -M:poly test` - test fails

```
clojure.lang.ExceptionInfo: Error while evaluating form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) in class-loader. Cause: java.lang.reflect.InvocationTargetException; Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote; clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)} {:form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote )))}
        at polylith.clj.core.test_runner_orchestrator.core$__GT_eval_in_project$fn__11005.invoke(core.clj:110)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements$fn__40531.invoke(core.clj:59)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements.invokeStatic(core.clj:58)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements.invoke(core.clj:52)
        at polylith.clj.core.clojure_test_test_runner.core$create$reify__40558.run_tests(core.clj:98)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project_with_test_runner.invokeStatic(core.clj:88)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project_with_test_runner.invoke(core.clj:77)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project.invokeStatic(core.clj:159)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project.invoke(core.clj:129)
        at polylith.clj.core.test_runner_orchestrator.core$run.invokeStatic(core.clj:214)
        at polylith.clj.core.test_runner_orchestrator.core$run.invoke(core.clj:197)
        at polylith.clj.core.test_runner_orchestrator.interface$run.invokeStatic(interface.clj:5)
        at polylith.clj.core.test_runner_orchestrator.interface$run.invoke(interface.clj:4)
        at polylith.clj.core.command.test$run.invokeStatic(test.clj:11)
        at polylith.clj.core.command.test$run.invoke(test.clj:5)
        at polylith.clj.core.command.core$execute.invokeStatic(core.clj:96)
        at polylith.clj.core.command.core$execute.invoke(core.clj:71)
        at polylith.clj.core.command.interface$execute_command.invokeStatic(interface.clj:5)
        at polylith.clj.core.command.interface$execute_command.invoke(interface.clj:4)
        at polylith.clj.core.poly_cli.core$_main.invokeStatic(core.clj:33)
        at polylith.clj.core.poly_cli.core$_main.doInvoke(core.clj:7)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.main$main_opt.invokeStatic(main.clj:514)
        at clojure.main$main_opt.invoke(main.clj:510)
        at clojure.main$main.invokeStatic(main.clj:664)
        at clojure.main$main.doInvoke(main.clj:616)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.main.main(main.java:40)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at polylith.clj.core.common.class_loader$invoke_in_STAR_.invokeStatic(class_loader.clj:31)
        at polylith.clj.core.common.class_loader$invoke_in_STAR_.doInvoke(class_loader.clj:27)
        at clojure.lang.RestFn.invoke(RestFn.java:494)
        at polylith.clj.core.common.class_loader$eval_in_STAR_$print_read_eval__3811.invoke(class_loader.clj:49)
        at polylith.clj.core.common.class_loader$eval_in_STAR_.invokeStatic(class_loader.clj:51)
        at polylith.clj.core.common.class_loader$eval_in_STAR_.invoke(class_loader.clj:45)
        at polylith.clj.core.common.class_loader$eval_in.invokeStatic(class_loader.clj:61)
        at polylith.clj.core.common.class_loader$eval_in.invoke(class_loader.clj:60)
        at polylith.clj.core.common.interface$eval_in.invokeStatic(interface.clj:22)
        at polylith.clj.core.common.interface$eval_in.invoke(interface.clj:21)
        at polylith.clj.core.test_runner_orchestrator.core$__GT_eval_in_project$fn__11005.invoke(core.clj:108)
        ... 30 more
Caused by: Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7132)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.analyze(Compiler.java:6762)
        at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3900)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7126)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.analyze(Compiler.java:6762)
        at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:6137)
        at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5479)
        at clojure.lang.Compiler$FnExpr.parse(Compiler.java:4041)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7122)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.eval(Compiler.java:7191)
        at clojure.lang.Compiler.eval(Compiler.java:7183)
        at clojure.lang.Compiler.eval(Compiler.java:7149)
        at clojure.core$eval.invokeStatic(core.clj:3215)
        at clojure.core$eval.invoke(core.clj:3211)
        at clojure.core$eval73128.invokeStatic(NO_SOURCE_FILE:0)
        at clojure.core$eval73128.invoke(NO_SOURCE_FILE)
        at clojure.lang.Compiler.eval(Compiler.java:7194)
        at clojure.lang.Compiler.eval(Compiler.java:7149)
        ... 45 more
Caused by: clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)}
        at clojure.lang.Compiler$ConstantExpr$Parser.parse(Compiler.java:2016)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7124)
        ... 65 more
Couldn't run test statement for the app project: (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) clojure.lang.ExceptionInfo: Error while evaluating form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) in class-loader. Cause: java.lang.reflect.InvocationTargetException; Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote; clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)} {:form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote )))}

Test results:  passes,  failures,  errors.
```

**Comment by tengstrand** [comment]
Okay, thanks, will have a look!

**Comment by ieugen** [comment]
To clarify, I used the latest commit `9053b190d5f3b0680ac4fe5c5f1851f7c0d40830` as of today.

**Comment by tengstrand** [comment]
It was hard for me to reproduce this.
Is it possible for you to set up a small repository that reproduces this problem and share it?

If you run the `check` command, do you get a "Warning 206: Unreadable namespace"?

**Comment by ieugen** [comment]
hi @tengstrand , 

I can try to make a repro this week.

```
$ clojure -M:poly check
....
  Warning 206: Unreadable namespace in app: bases/app/src/dre/app/config.clj. To ignore this warning, execute 'poly help check' and follow the instructions for warning 206.
....

```

**Comment by tengstrand** [comment]
If you ignore the file, does the problem go away (I guess it will not)?

**Comment by ieugen** [comment]
Hi @tengstrand , 
The file is part of the codebase. 
It contains default configuration options.
We can't ignore it or I don't know what you mean by ignoring it. 

In my opinion, the file, being a clojure file on a path, should have a namespace definition.
I remember asking about this and the consensus was that it's not required. 

So the code should work, issue a warning.
Right now it fails with a very strange exception for which the cause is not easy to track down.

**Comment by tengstrand** [comment]
If you run `help check` and read about warning 206, it tells you how you can get rid of that warning.
When I think about it, that will not solve the problem, only get rid of the warning, so I still need that repository that can reproduce the problem, if that is possible. Thanks.

**Comment by ieugen** [comment]
Please check out this branch: https://github.com/ieugen/poly-rcf/tree/poly-279  .
Run `clojure -M:poly test` .

**Comment by tengstrand** [comment]
Thanks @ieugen! 

I had a quick look, but it wasn't obvious to me how to fix this. I'm working on other issues right now, so I have instead asked @furkan3ayraktar to have a look at this, when he is back from his vacation, because he knows things like this!

**Comment by furkan3ayraktar** [comment]
Late to the party, but I'd be happy to help.

Is this `config.clj` a standard configuration file for a library/framework you use, or is it something you chose? I haven't seen `.clj` files used as configuration files; it's usually `.edn` files used as configuration. If using `.clj` files as configuration files is a common practice in a specific library/framework, it would be great if you could direct me there.

Polylith treats `.clj` files as source files and parses them for validation. In the [default test runner](https://github.com/polyfy/polylith/tree/master/components/clojure-test-test-runner/src/polylith/clj/core/clojure_test_test_runner), the poly tool finds all the namespaces under the test alias and runs them through `clojure.test`. If this config file is included under the test alias, the default test runner will pass it to the test runner like this:
https://github.com/polyfy/polylith/blob/9053b190d5f3b0680ac4fe5c5f1851f7c0d40830/components/clojure-test-test-runner/src/polylith/clj/core/clojure_test_test_runner/core.clj#L7-L11

When it's required by `(require '~ns-symbol)`, it will throw an exception since the `config.clj` is not a valid namespace. One option to solve this within the poly tool's scope is that the default test runner can ignore the files that are listed under [`:ignore-files`](https://github.com/polyfy/polylith/blob/9053b190d5f3b0680ac4fe5c5f1851f7c0d40830/scripts/output/help/02-check.txt#L73-L81C11) setting. You can then add your `config.clj` as an ignored file, and the poly tool can skip processing that file (I haven't tested this with your example repository, but could you test adding your `config.clj` under `:ignore-files` and see if it solves the issue? Maybe it already does what we want.). Another option could be extending the default test runner and [writing your own test runner](https://polylith.gitbook.io/poly/workflow/testing#test-runners) to run tests. 

We follow the best practices from the community as we cannot support everyone's coding style, although we try to adapt if there is a need or request from the community. These practices change over time, and we try evolving Polylith accordingly.

**Comment by tengstrand** [comment]
The support for `:ignore-files` is not part of the latest 0.2.17.alpha release, but lives in the master branch right now.

The `:ignore-files` only suppress' warnings, but it doesn't exclude the files when running tests.
This requires a small redesign before ending up in the 0.2.18 release, and I need to take a closer look at how this can be configured.

I think we need a separation between suppressing warnings and/or ignoring files in tests. Will create an issue later today when I'm back home. Will let you know when it has been implemented, so you can try it out.

**Comment by ieugen** [comment]
@furkan3ayraktar : thanks for the details. 
I don't mind the test runner to fail for these cases. 
IMO `clj` files should have a proper ns. 

However, it took me a LOT of time to find the cause of this error and I was not the one that found the issue.
I believe  better error message for this case will be enough. 
`:ignore-files` is also a good option - but only after you know the cause. 
In my case, I know it, but the next guy might lose a couple of hours hunting it down.

**Comment by tengstrand** [comment]
If the workspace has a warning 206 "Unreadable namespace in brick/project" and if the tests fail, then that warning will be displayed after the exception. Please have a look @ieugen!

**Comment by ieugen** [comment]
hi @tengstrand ,

I ran using this commit f85a99a6c446a6808e12d051359bc319fab86769 . 

It seems the warning is there.  

```
clojure.lang.ExceptionInfo: Error while evaluating form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) in class-loader. Cause: java.lang.reflect.InvocationTargetException; Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote; clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)} {:form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote )))}
        at polylith.clj.core.test_runner_orchestrator.core$__GT_eval_in_project$fn__11417.invoke(core.clj:110)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements$fn__41636.invoke(core.clj:59)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements.invokeStatic(core.clj:58)
        at polylith.clj.core.clojure_test_test_runner.core$run_test_statements.invoke(core.clj:52)
        at polylith.clj.core.clojure_test_test_runner.core$create$reify__41663.run_tests(core.clj:98)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project_with_test_runner.invokeStatic(core.clj:88)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project_with_test_runner.invoke(core.clj:77)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project.invokeStatic(core.clj:159)
        at polylith.clj.core.test_runner_orchestrator.core$run_tests_for_project.invoke(core.clj:129)
        at polylith.clj.core.test_runner_orchestrator.core$run.invokeStatic(core.clj:221)
        at polylith.clj.core.test_runner_orchestrator.core$run.invoke(core.clj:203)
        at polylith.clj.core.test_runner_orchestrator.interface$run.invokeStatic(interface.clj:5)
        at polylith.clj.core.test_runner_orchestrator.interface$run.invoke(interface.clj:4)
        at polylith.clj.core.command.test$run.invokeStatic(test.clj:11)
        at polylith.clj.core.command.test$run.invoke(test.clj:5)
        at polylith.clj.core.command.core$execute.invokeStatic(core.clj:113)
        at polylith.clj.core.command.core$execute.invoke(core.clj:84)
        at polylith.clj.core.command.interface$execute_command.invokeStatic(interface.clj:5)
        at polylith.clj.core.command.interface$execute_command.invoke(interface.clj:4)
        at polylith.clj.core.poly_cli.core$_main.invokeStatic(core.clj:33)
        at polylith.clj.core.poly_cli.core$_main.doInvoke(core.clj:7)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.main$main_opt.invokeStatic(main.clj:514)
        at clojure.main$main_opt.invoke(main.clj:510)
        at clojure.main$main.invokeStatic(main.clj:664)
        at clojure.main$main.doInvoke(main.clj:616)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.main.main(main.java:40)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at polylith.clj.core.common.class_loader$invoke_in_STAR_.invokeStatic(class_loader.clj:31)
        at polylith.clj.core.common.class_loader$invoke_in_STAR_.doInvoke(class_loader.clj:27)
        at clojure.lang.RestFn.invoke(RestFn.java:494)
        at polylith.clj.core.common.class_loader$eval_in_STAR_$print_read_eval__3876.invoke(class_loader.clj:49)
        at polylith.clj.core.common.class_loader$eval_in_STAR_.invokeStatic(class_loader.clj:51)
        at polylith.clj.core.common.class_loader$eval_in_STAR_.invoke(class_loader.clj:45)
        at polylith.clj.core.common.class_loader$eval_in.invokeStatic(class_loader.clj:61)
        at polylith.clj.core.common.class_loader$eval_in.invoke(class_loader.clj:60)
        at polylith.clj.core.common.interface$eval_in.invokeStatic(interface.clj:27)
        at polylith.clj.core.common.interface$eval_in.invoke(interface.clj:26)
        at polylith.clj.core.test_runner_orchestrator.core$__GT_eval_in_project$fn__11417.invoke(core.clj:108)
        ... 30 more
Caused by: Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7132)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.analyze(Compiler.java:6762)
        at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3900)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7126)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.analyze(Compiler.java:6762)
        at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:6137)
        at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5479)
        at clojure.lang.Compiler$FnExpr.parse(Compiler.java:4041)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7122)
        at clojure.lang.Compiler.analyze(Compiler.java:6806)
        at clojure.lang.Compiler.eval(Compiler.java:7191)
        at clojure.lang.Compiler.eval(Compiler.java:7183)
        at clojure.lang.Compiler.eval(Compiler.java:7149)
        at clojure.core$eval.invokeStatic(core.clj:3215)
        at clojure.core$eval.invoke(core.clj:3211)
        at clojure.core$eval73727.invokeStatic(NO_SOURCE_FILE:0)
        at clojure.core$eval73727.invoke(NO_SOURCE_FILE)
        at clojure.lang.Compiler.eval(Compiler.java:7194)
        at clojure.lang.Compiler.eval(Compiler.java:7149)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 42 more
Caused by: clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)}
        at clojure.lang.Compiler$ConstantExpr$Parser.parse(Compiler.java:2016)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7124)
        ... 63 more
Couldn't run test statement for the app project: (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) clojure.lang.ExceptionInfo: Error while evaluating form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote ))) in class-loader. Cause: java.lang.reflect.InvocationTargetException; Syntax error compiling quote at (0:0).
Wrong number of args (0) passed to quote; clojure.lang.ExceptionInfo: Wrong number of args (0) passed to quote {:form (quote)} {:form (do (clojure.core/use (quote clojure.test)) (clojure.core/require (quote )) (clojure.test/run-tests (quote )))}

  Warning 206: Unreadable namespace in app: bases/app/src/dre/app/config.clj. To ignore this warning, execute 'poly help check' and follow the instructions for warning 206.

Test results:  passes,  failures,  errors.
```

**Comment by ieugen** [comment]
I think we can close the issue. 
We get a warning. The message is not a clear indication of the issue. 
So if a user gets more warnings then he might ignore this one, but at least it's there.
@tengstrand  Can it be an error ? Because it does behave like an error => build stops with error. 

Thank you for checking this out :)

**Comment by tengstrand** [comment]
It sunds reasonable. Do you have an opinion @furkan3ayraktar ?

**Comment by furkan3ayraktar** [comment]
@tengstrand Warning 206 can instead be an error. Having an unreadable namespace in the workspace (if it's not ignored intentionally) is an error rather than a warning to me.

**Comment by tengstrand** [comment]
I have turned the warning 206 into an error 111. Please have a look @ieugen.

**Comment by ieugen** [comment]
hi @tengstrand , I did the following:
-  upgrade poly sha to latest commit
- ran poly test - all passed
- commented out ns declaration in file
- ran poly test again - fail with error 111

I did notice that I did not get the unusable/misleading/non-informative clojure stack traces .
This is a plus IMO 

```
 clojure -M:poly test
  Error 111: Unreadable namespace in app: bases/app/src/dre/app/config.clj. To ignore this error, execute 'poly help check' and follow the instructions for error 111.
```  

I am closing this issue since IMO it is done.
Thank you again for looking into this.

**Comment by tengstrand** [comment]
Sounds great!
